src/pages/trade/ui/order-form/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { useEffect } from 'react';
+import { observer } from 'mobx-react-lite';
+import { Button } from '@penumbra-zone/ui/Button';
+import { Text } from '@penumbra-zone/ui/Text';
+import { connectionStore } from '@/shared/model/connection';
+import { useBalances } from '@/shared/api/balances';
+import { usePathToMetadata } from '../../model/use-path';
+import { OrderInput } from './order-input';
+import { SegmentedControl } from './segmented-control';
+import { ConnectButton } from '@/features/connect/connect-button';
+import { Slider } from './slider';
+import { InfoRow } from './info-row';
+import { orderFormStore, Direction } from './order-form-store';
+
+const useOrderFormStore = () => {
+  const { baseAsset, quoteAsset } = usePathToMetadata();
+  const { data: balances } = useBalances();
+  const { setAssets, setBalances } = orderFormStore;
+
+  useEffect(() => {
+    if (baseAsset && quoteAsset) {
+      setAssets(baseAsset, quoteAsset);
+    }
+  }, [baseAsset, quoteAsset, setAssets]);
+
+  useEffect(() => {
+    if (balances) {
+      setBalances(balances);
+    }
+  }, [balances, setBalances]);
+
+  return orderFormStore;
+};
+
+export const OrderForm = observer(() => {
+  const { connected } = connectionStore;
+  const {
+    baseAsset,
+    quoteAsset,
+    direction,
+    setDirection,
+    submitOrder,
+    isLoading,
+    gasFee,
+    exchangeRate,
+  } = useOrderFormStore();
+
+  return (
+    <div>
+      <SegmentedControl direction={direction} setDirection={setDirection} />
+      <OrderInput
+        label={direction}
+        value={baseAsset.amount}
+        onChange={baseAsset.setAmount as (amount: string, ...args: unknown[]) => void}
+        min={0}
+        max={1000}
+        isEstimating={baseAsset.isEstimating}
+        isApproximately={baseAsset.isApproximately}
+        denominator={baseAsset.symbol}
+      />
+      <OrderInput
+        label={direction === Direction.Buy ? 'Pay with' : 'Receive'}
+        value={quoteAsset.amount}
+        onChange={quoteAsset.setAmount as (amount: string, ...args: unknown[]) => void}
+        isEstimating={quoteAsset.isEstimating}
+        isApproximately={quoteAsset.isApproximately}
+        denominator={quoteAsset.symbol}
+      />
+      <Slider steps={8} asset={quoteAsset} />
+      <div className='mb-4'>
+        <InfoRow
+          label='Trading Fee'
+          value='Free'
+          valueColor='success'
+          toolTip='On Penumbra, trading fees are completely free.'
+        />
+        {/* @TODO: Add gas fee */}
```

Comment:
can we create a separate issue for this? The scope should include integrating the existing balance selector logic from minifront to properly select the asset metadata for the gas fee, since we also support paying gas with alternative asset fees which complicates this logic. Additionally, the tooltip should provide a more descriptive message about gas fees.

<img width="264" alt="Screenshot 2024-11-19 at 10 08 35 PM" src="https://github.com/user-attachments/assets/679115ab-37fa-44c5-bcb5-7657954bf1eb">

Reply:
Still awaiting the proper texts for the tooltips.